File: 1482-minimum-number-of-days-to-make-m-bouquets/1482-minimum-number-of-days-to-make-m-bouquets.py
class Solution:
    def minDays(self, bloomDay: List[int], m: int, k: int) -> int:
        if m*k > len(bloomDay):
            return -1
        left = min(bloomDay)
        right = max(bloomDay)
        ans = right
        
        while left <= right:
            count = 0
            number = 0
            mid = (left+right)// 2
            for i in range(len(bloomDay)):
                if bloomDay[i] <= mid:
                    count += 1
                    if count == k:
                        number += 1
                        count = 0
                if bloomDay[i] > mid:
                    count = 0
            if number < m:
                left = mid+1
            if number >= m:
                right = mid - 1
                ans = mid
        return ans

File: 1482-minimum-number-of-days-to-make-m-bouquets/1482-minimum-number-of-days-to-make-m-bouquets.js
/**
 * @param {number[]} bloomDay
 * @param {number} m
 * @param {number} k
 * @return {number}
 */
var minDays = function(bloomDay, m, k) {
    if(m*k > bloomDay.length)return -1
    let start = 0,end = Number.MAX_VALUE,final=-1
    
    while(start <= end){
          let mid = Math.floor((start+end)/2)
          if(helper(bloomDay,m,k,mid)){
              end = mid - 1
              final = mid
          }else{
              start = mid+1
          }
    }
    return final
};

var helper = (days,m,k,mid)=>{
    let count=0,total=0
    for(let i=0;i<days.length;i++){
        if(days[i] <= mid){
            count++
            if(count == k){
                total++
                count = 0
            }
        }else{
            count = 0
        }
    }
    return total >= m
}